src/goDebugConfiguration.ts

```diff
@@ -70,8 +70,14 @@ export class GoDebugConfigurationProvider implements vscode.DebugConfigurationPr
 		});
 
 		const dlvConfig: { [key: string]: any } = goConfig.get('delveConfig');
-		if (!debugConfiguration.hasOwnProperty('useApiV1') && dlvConfig.hasOwnProperty('useApiV1')) {
-			debugConfiguration['useApiV1'] = dlvConfig['useApiV1'];
+		let useApiV1 = false;
+		if (debugConfiguration.hasOwnProperty('useApiV1')) {
+			useApiV1  = debugConfiguration['useApiV1'] === true;

```

Comment:
nit: This and the line below can also be `useApiV1 = !!debugConfiguration['useApiV1']`.